### ISSUE TITLE
[6.0.0.rc-3] redux-form/CHANGE alters field-value to field-name in conjunction with React DnD

### ISSUE THREAD

**Issue by peletiah** [erikras/redux-form#1352]
I'm using redux-form 6.0.0.rc-3 in conjunction with [React DnD](http://gaearon.github.io/react-dnd/) and I noticed a weird behaviour when I focus an input and slightly drag on it without really changing anything. When I do that, `redux-form/CHANGE` is called and its payload is the field-name, in consequence the field-value is overwritten by the field-name. 

None of the DnD-reducers are called by this small drag, since the criteria for a full move action have not been triggered (See `sequenceTarget`-definition further down).

The issue does not appear when I drag the input a bit further out or click and drag outside of the input. 
Depending on how far to the left or right of the input I place the mouse and which direction I drag, the input right or left of the focused input is modified. Here's a screencast to illustrate the issue:

![](http://i.giphy.com/l0HlAWgBEurlybhsc.gif)

This is based on [this code](https://github.com/peletiah/goatFS-Client/tree/29ccabec3ca0ff57d9eedab8bcf7cbda3da2b373/scripts/routing). 

I'm loading an object with fetch and put it in `routeState` in my redux-store:

``` json
{"sequences":[
  {
    "command":"log",
    "sequence":1,
    "data":"INFO Calling Extension 200"
  },
  {
    "command":"set",
    "sequence":2,
    "data":"effective_caller_id_name=John"
  },
  {
    "command":"bridge",
    "sequence":3,
    "data":"user/200"
  }
]}
```

The basic redux-form and `FieldArray` is set up like this (abbreviated, from [Routing.js](https://github.com/peletiah/goatFS-Client/blob/29ccabec3ca0ff57d9eedab8bcf7cbda3da2b373/scripts/routing/Routing.js)):

``` javascript

const renderSequences = ({ fields, moveSequence }) => (
  <div>
    {fields.map((sequenceField, index) =>
      <Sequence key={ `${sequenceField}.sequence` }
                index={ index }
                moveSequence ={ moveSequence }
                sequenceField={ sequenceField } />
    )}
  </div>
)

@DragDropContext(HTML5Backend)
class Routing extends Component {
render() {
    const { sequences } = this.props

    return (
        <FieldArray name="sequences"
            component   =  { renderSequences }
            moveSequence = { this.moveSequence }
        />
    );
  }
};

Routing = reduxForm({
    form: 'sequenceform'
  }
)(Routing)

Routing = connect(
  state => ({
    initialValues: {sequences:state.routeState.sequences}
  })
)(Routing)
```

Then I render the `Fields` in [`<Sequence/>`](https://github.com/peletiah/goatFS-Client/blob/29ccabec3ca0ff57d9eedab8bcf7cbda3da2b373/scripts/routing/Sequence.js) like this:

``` javascript
const renderField = field => (
  <span>
      <input {...field.input}/>
      {field.touched && field.error && <span>{field.error}</span>}
  </span>
)

const renderSpan = field => (
  <span>
    {field.input.value}
  </span>
)

const sequenceTarget = {
  hover(props, monitor, component) {
    const dragIndex = monitor.getItem().index;
    const hoverIndex = props.index;
    if (dragIndex === hoverIndex) {
      return;
    }
    const hoverBoundingRect = findDOMNode(component).getBoundingClientRect();
    const hoverMiddleY = (hoverBoundingRect.bottom - hoverBoundingRect.top) / 2;
    const clientOffset = monitor.getClientOffset();
    const hoverClientY = clientOffset.y - hoverBoundingRect.top;
    if (dragIndex < hoverIndex && hoverClientY < hoverMiddleY) {
      return;
    }
    if (dragIndex > hoverIndex && hoverClientY > hoverMiddleY) {
      return;
    }
    props.moveSequence(dragIndex, hoverIndex);
    monitor.getItem().index = hoverIndex;
  }
}

@DropTarget(ItemTypes.SEQUENCE, sequenceTarget, connect => ({
  connectDropTarget: connect.dropTarget()
}))
@DragSource(ItemTypes.SEQUENCE, sequenceSource, (connect, monitor) => ({
  connectDragSource: connect.dragSource(),
  isDragging: monitor.isDragging()
}))
class Sequence extends Component {
  static propTypes = {
    connectDragSource: PropTypes.func.isRequired,
    connectDropTarget: PropTypes.func.isRequired,
    index: PropTypes.number.isRequired,
    isDragging: PropTypes.bool.isRequired,
    sequenceField: PropTypes.any.isRequired,
    moveSequence: PropTypes.func.isRequired
  };

  render() {
    const { sequenceField,
            isDragging,
            connectDragSource,
            connectDropTarget
          } = this.props

    return connectDragSource(connectDropTarget(
      <div>
        <Field
          name={`${sequenceField}.sequence`}
          type="text"
          component={renderSpan} />
        <div>
          <Field
            name={`${sequenceField}.command`}
            type="text"
            component={renderField}
            placeholder="command"/>
          <Field
            name={`${sequenceField}.data`}
            type="text"
            component={renderField}
            placeholder="data"/>
        </div>
      </div>
    ));
  }
}

export default Sequence
```

I hope I have not missed clues that I'm causing this issue through my own code. But since none of my own reducers are triggered and React DnD does not seem to be directly involved, I dare to believe that there might be an issue in redux-form.

Also summoning @gaearon!

**Comment by peletiah** [comment]
Still present in rc.5, but now the form is emptied instead.

**Comment by jwyuen** [comment]
+1 I'm having this issue too on redux-form 6.2.0 and react-dnd 2.1.4

**Comment by Marinolinderhof** [comment]
my work around is to disable the fields when begin to drag.

**Comment by paulhands** [comment]
We had a similar issue, and managed to track the problem to trying to drag a field onto another field. We wanted the component to disappear while it was being dragged, and therefore found that using

`    const visibility = isDragging ? 'hidden' : 'visible';`

and

`<DRAGGABLECOMPONENT style={{ visibility }} />`

inside the connectDragSource(connectDropTarget( etc. removed the problem.

**Comment by peletiah** [comment]
Appears to be fixed in 6.5.0
